kyuubi-hive-beeline/src/main/scala/org/apache/kyuubi/beeline/BeelineConf.scala

```diff
@@ -0,0 +1,189 @@
+      .doc("Name of the Kerberos principal.")
+      .version("1.7.0")
+      .stringConf
+      .createWithDefault("")
+
+  val BEELINE_HA_ADDRESSES: ConfigEntry[String] = buildConf("kyuubi.beeline.ha.addresses")
+    .doc("The connection string for the discovery ensemble.")
+    .version("1.7.0")
+    .stringConf
+    .createWithDefault("")
+
+  val BEELINE_HA_NAMESPACE: ConfigEntry[String] = buildConf("kyuubi.beeline.ha.namespace")
+    .doc("The root directory for the service to deploy its instance uri.")
+    .version("1.7.0")
+    .stringConf
+    .createWithDefault("kyuubi")
+
+  val BEELINE_USE_SSL: ConfigEntry[Boolean] = buildConf("kyuubi.beeline.use.SSL")
+    .doc("Set this to true for using SSL encryption.")
+    .version("1.7.0")
+    .booleanConf
+    .createWithDefault(false)
+
+  val BEELINE_SSL_TRUSTSTORE: OptionalConfigEntry[String] =
+    buildConf("kyuubi.beeline.ssl.truststore")
+      .doc("The path to the SSL TrustStore.")
+      .version("1.7.0")
+      .stringConf
+      .createOptional
+
+  val BEELINE_SSL_TRUSTSTORE_PASSWORD: OptionalConfigEntry[String] =
+    buildConf("kyuubi.beeline.ssl.truststore.password")
+      .doc("The password to the SSL TrustStore.")
+      .version("1.7.0")
+      .stringConf
+      .createOptional
+
+  val BEELINE_SASL_QOP: ConfigEntry[String] = buildConf("kyuubi.beeline.sasl.qop")
+    .doc("Sasl QOP enable higher levels of protection for Kyuubi communication with " +
+      "clients.<ul>" +
+      " <li>auth - authentication only (default)</li>" +
+      " <li>auth-int - authentication plus integrity protection</li>" +
+      " <li>auth-conf - authentication plus integrity and confidentiality protection. This is" +
+      " applicable only if Kyuubi is configured to use Kerberos authentication.</li> </ul>")
+    .version("1.7.0")
+    .stringConf
+    .checkValues(SaslQOP.values.map(_.toString))
+    .transform(_.toLowerCase(Locale.ROOT))
+    .createWithDefault(SaslQOP.AUTH.toString)
+
+  val BEELINE_SESSION_CONFS: OptionalConfigEntry[String] =
+    buildConf("kyuubi.beeline.session.confs")
+      .doc("Optional `Semicolon(;)` separated `key=value` parameters for the JDBC/ODBC driver." +
+        " Such as `user`, `password` and `hive.server2.proxy.user`.")
+      .version("1.7.0")
+      .stringConf
+      .createOptional
+
+  val BEELINE_KYUUBI_CONFS: OptionalConfigEntry[String] =
+    buildConf("kyuubi.beeline.kyuubi.confs")
+      .doc("Optional `Semicolon(;)` separated `key=value` parameters for Kyuubi server" +
+        " to create the corresponding engine, dismissed if engine exists.")
+      .version("1.7.0")
+      .stringConf
+      .createOptional
+
+  val BEELINE_KYUUBI_VARS: OptionalConfigEntry[String] =
+    buildConf("kyuubi.beeline.kyuubi.vars")
+      .doc("Optional `Semicolon(;)` separated `key=value` parameters for" +
+        " Spark/Hive variables used for variable substitution.")
+      .version("1.7.0")
+      .stringConf
+      .createOptional
```

Comment:
It is better to support to specify the config with prefix.
Such as :

```
kyuubi.beeline.sessionConf.
kyuubi.beeline.kyuubiConf.
kyuubi.beeline.kyuubiVar.
```

You can refer: KyuubiConf::KYUUBI_ENGINE_ENV_PREFIX

Reply:
For example:
```
kyuubi.beeline.sessionConf.hive.server2.proxy.user=b_proxy
kyuubi.beeline.kyuubiConf.kyuubi.engine.share.level=CONNECTION
kyuubi.beeline.kyuubiVar.spark.sql.shuffle.partitions=1000
```